src/game/behaviors/sparkle_spawn_star.inc.c

```diff
@@ -36,11 +36,15 @@ void set_sparkle_spawn_star_hitbox(void) {
     }
 }
 
-void set_home_to_mario(void) {
+void spawned_star_set_target_above_mario(void) {
     vec3f_copy_y_off(&o->oHomeVec, &gMarioObject->oPosVec, 250.0f);
 
-    o->oPosY = o->oHomeY;
+    // Check that the star isn't clipping inside the ceiling
+    if (o->oHomeY > (gMarioState->ceilHeight - 50)) {
```

Comment:
(Not reviewing with an IDE in front of me so can't check myself.) How does ceilHeight work, is the value always going to be greater than Mario's Y position? What happens if there is no ceiling for example?

Reply:
> (Not reviewing with an IDE in front of me so can't check myself.) How does ceilHeight work, is the value always going to be greater than Mario's Y position? What happens if there is no ceiling for example?

Apparently `ceilHeight` is always `20000` when Mario has no ceilings. Don't really want to deal with this so I made it so that it only does this if `gMarioState->ceil` isn't `NULL`. So whatever potential side effects this may have are now solved.